client/src/controllers/ProgressController.test.js

```diff
@@ -94,4 +96,25 @@ describe('ProgressController', () => {
     expect(button.getAttribute('disabled')).toBeNull();
     expect(button.classList.contains('button-longrunning-active')).toBe(false);
   });
+
+  it('should return to the original state when deactivate is called', async () => {
+    const button = document.querySelector('.button-longrunning');
+    const label = document.querySelector('#em-el');
+    const controller = application.getControllerForElementAndIdentifier(
+      button,
+      'w-progress',
+    );
+
+    button.click();
+    jest.advanceTimersByTime(10);
+    await new Promise(queueMicrotask);
+
+    expect(label.textContent).toBe('Loading');
+    expect(button.classList.contains('button-longrunning-active')).toBe(true);
+
+    controller.deactivate();
+    await new Promise(queueMicrotask);
+    expect(label.textContent).toBe('Sign in');
+    expect(button.classList.contains('button-longrunning-active')).toBe(false);
```

Comment:
Would be nice to also add assertions for the `disabled` attribute, and maybe assertions that `clearTimeout` is called with the correct timeout.